src/monitors/CoverageMonitor.v3

```diff
@@ -56,28 +67,101 @@ class CoverageMonitor extends Monitor {
 		var tracer = InstrTracer.new(), d = DataReader.new([]);
 		var blockCovered = false;
 		var it = BytecodeIterator.new();
-
-		for (j < entries.length) {
-			var e = entries[j];
-			Trace.OUT.put1("func %q:", e.func.render(e.module.names, _)).outln();
-			for (bi = it.reset(e.func); bi.more(); bi.next()) {
-				bi.current();
-				d.reset(e.func.orig_bytecode, bi.pc, e.func.orig_bytecode.length);
-
-				if (inBasicBlockMode) {
-					if (isStartOfBlock[e.start + bi.pc]) {
-						blockCovered = executed[e.start + bi.pc];
-						Trace.OUT.puts("->");
+		if (inCodeMode || !inBasicBlockMode) {
+			for (j < entries.length) {
+				var e = entries[j];
+				Trace.OUT.puts(TermColors.CYAN);
+				Trace.OUT.put1("func %q:", e.func.render(e.module.names, _)).outln();
```

Comment:
To shorten this up a bit, you can make `Trace.OUT` into a local or global variable, and also do chaining, so that you have something like:

```
out.puts(...)
     .puts(...)
     .puts(...)
```